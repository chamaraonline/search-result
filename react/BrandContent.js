import React, { Component } from 'react'
import { compose, equals, find, prop } from 'ramda'
import { Query } from 'react-apollo'
import getBrandContent from './queries/getBrandContent.gql'
import getBrandList from './queries/getBrandList.gql'
import { Spinner, Tab, Tabs } from 'vtex.styleguide'
import ProductShelfContent from './components/ProductShelfContent'
import { VideoSlider } from 'vtex.video-slider'
import styles from './searchResult.css'

class BrandContent extends Component {
  constructor() {
    super()
    this.state = {
      currentTab: 1,
    }
    this.handleTabChange = this.handleTabChange.bind(this)
  }

  handleTabChange(tabIndex) {
    this.setState({
      currentTab: tabIndex,
    })
  }

  render() {
    const { brandSlug, summary, mobileLayoutMode } = this.props
    return (
      <Query query={getBrandList}>
        {({ data: { brands }, brandsLoading, brandListError }) => {
          if (brandsLoading || !brands) return <Spinner />
          if (brandListError) return <p>ERROR</p>

          const brand = find(
            compose(
              equals(brandSlug),
              prop('slug')
            ),
            brands
          )
          return (
            <Query
              query={getBrandContent}
              variables={{
                acronym: 'BrandContent',
                fields: ['brandId', 'description', 'newProducts', 'videos'],
                where: `(brandId=${brand.id})`,
                schema: 'brand-content-schema-v1',
              }}
            >
              {({ data, brandContentLoading, brandContentError }) => {
                if (brandContentLoading) return <Spinner />
                if (brandContentError) return <p>ERROR</p>

                const brandContent = {
                  id: '27c964bc-7639-11e9-828f-d599ef464471',
                  brandId: '2000000',
                  description: 'Test description',
                  newProducts: [
                    {
                      productId: '24',
                      description: 'Test product 24 description',
                    },
                    {
                      productId: '17',
                      description: 'Test product 17 description',
                    },
                  ],
                  videos: [
                    {
                      title: 'Video 1',
                      url:
                        'https://player.vimeo.com/video/142175573?title=0&byline=0&portrait=0',
                    },
                    {
                      title: 'Video 2',
                      url:
                        'https://player.vimeo.com/video/166034292?title=0&byline=0&portrait=0',
                    },
                  ],
                }
                return (
                  <div className={`${styles.brandGalleryContainer} bb b--muted-4`}>
                    <Tabs>
                      <Tab
                        label="New Products"
                        active={this.state.currentTab === 1}
                        onClick={() => this.handleTabChange(1)}
                      >
                        <div
                          className={`${
                            styles.brandGalleryProductShelf
                          } flex flex-row flex-wrap items-stretch bn ph1 pl9-l na4`}
                        >
                          {brandContent.newProducts &&
                            brandContent.newProducts.map(productItem => (
                              <ProductShelfContent
                                productId={productItem.productId}
                                key={`product-content-${productItem.productId}`}
                                summary={summary}
                                mobileLayoutMode={mobileLayoutMode}
                              />
                            ))}
                        </div>
                      </Tab>
                      <Tab
                        label="Videos"
                        active={this.state.currentTab === 2}
                        onClick={() => this.handleTabChange(2)}
                      >
                        <div className={styles.brandGalleryVideos}>
                          {brandContent.videos && (
                            <VideoSlider
                              videos={brandContent.videos.map(video => ({
                                videoUrl: video.url,
                                videoTitle: video.title,
                              }))}
                            />
                          )}
                        </div>
                      </Tab>
                      <Tab
                        label="About"
                        active={this.state.currentTab === 3}
                        onClick={() => this.handleTabChange(3)}
                      >
                        <div className={`${styles.brandGalleryDescription} w-100 mh9`}>
                          {brandContent.description && (
                            <p className={styles.brandGalleryDescriptionTextContent}>{brandContent.description}</p>
                          )}
                        </div>
                      </Tab>
                    </Tabs>
                  </div>
                )
              }}
            </Query>
          )
        }}
      </Query>
    )
  }
}

export default BrandContent
