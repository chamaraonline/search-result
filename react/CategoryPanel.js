import React, { Fragment } from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import { Link } from 'vtex.render-runtime'
import { facetOptionShape } from './constants/propTypes'
import searchResult from './searchResult.css'
import CategoriesHighlights from 'vtex.store-components/CategoriesHighlights'
import { Query } from 'react-apollo'
import GET_FACET_IMAGES from './queries/facetImagesByFacetIds.graphql'
import { path } from 'ramda'

const renderCategoryShelf = (category, noOfChildren, quantityOfItemsPerRow) => {

  const categoriesHighlighted = category.children
    .slice(0, noOfChildren)
    .map(child => ({
      id: child.id,
      name: child.name,
      to: child.link,
    }))
  const headerClasses = classNames(
    searchResult.categoryPanelHeaderRow,
    'flex flex-row flex-wrap items-stretch pa3 bn mh4-ns'
  )
  const itemClasses = classNames(
    searchResult.categoryPanelItemRow,
    'flex flex-row flex-wrap items-stretch pa3 bn mh4-ns'
  )
  let i = 0
  let facetIds = []
  const items = {}
  for (i = 0; i < categoriesHighlighted.length; i++) {
    items[`item-${categoriesHighlighted[i].id}`] = categoriesHighlighted[i]
    facetIds.push(categoriesHighlighted[i].id)
  }

  const variables = {facetIds: facetIds.join(","), facetType:"category", page: 1, pageSize: facetIds.length + 1}

  return (
    <Fragment key={`parent-fragment-${category.id}`}>
      <h4 className={`t-heading-4 ${headerClasses}`}>
        <Link className="link c-muted-1" to={category.link}>{category.name}</Link>
      </h4>
      {facetIds.length !== 0 && (
        <Query
          query={GET_FACET_IMAGES}
          variables={variables}
        >
          {({ loading, error, data }) => {
            if (loading || error) return <div/>

            const categories = path(['facetImagesByFacetIds'], data)
            categories.forEach(category => {
              let key = `item-${category.facetId}`
              if(Object.keys(items).includes(key)){
                items[key] = { ...items[key], ...{ image: category.imageUrl} }
              }
            })

            return (
              <div className={itemClasses}>
                <CategoriesHighlights
                  categoriesHighlighted={items}
                  showCategoriesHighlighted
                  quantityOfItems={quantityOfItemsPerRow}
                />
              </div>
            )
          }}
        </Query>
      )}
      {category.children.map(child => {
        if (child.children && child.children.length > 0) {
          return renderCategoryShelf(child, noOfChildren, quantityOfItemsPerRow)
        }
      })}
    </Fragment>
  )
}

const CategoryPanel = ({ tree, noOfCategories, quantityOfItemsPerRow }) => {
  return (
    <div>
      {tree.map(category => {
        return renderCategoryShelf(
          category,
          noOfCategories,
          quantityOfItemsPerRow
        )
      })}
    </div>
  )
}

CategoryPanel.propTypes = {
  /** Category tree */
  tree: PropTypes.arrayOf(facetOptionShape),
  /** Number of category filters to be shown */
  noOfCategories: PropTypes.number,
  /** Number of categories per row */
  quantityOfItemsPerRow: PropTypes.number,
}

CategoryPanel.defaultProps = {
  tree: [],
  noOfCategories: 1000,
  quantityOfItemsPerRow: 6,
}

export default CategoryPanel
