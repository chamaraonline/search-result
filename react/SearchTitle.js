import classNames from 'classnames'
import React from 'react'
import { zip } from 'ramda'

import styles from './searchResult.css'

const isIterable = object =>
  object != null && typeof object[Symbol.iterator] === 'function'

const getClusterById = (products, id) => {
  if (!isIterable(products)) {
    return
  }

  for (const product of products) {
    if (!product || !isIterable(product.productClusters)) {
      continue
    }
    for (const cluster of product.productClusters) {
      if (cluster.id === id) {
        return cluster.name
      }
    }
  }
}

const padRight = (array, num) =>
  array.concat(Array.from(Array(num))).slice(0, num)

const SearchTitle = ({ params, map, products }) => {
  // Terrible hotfix for product clusters/collections name below!
  // The title should come from store-graphql. Once that's done,
  // all this logic should be removed from here.

  let title = null

  // matches title "sources" with its respective map,
  // and gets the last valid one
  const categories = [
    params.department,
    params.category,
    params.subcategory,
    params.term,
  ]

  const splitMap = padRight(map ? map.split(',') : [], categories.length)

  const titleSources = zip(categories, splitMap).reverse()

  const [titleValue, titleMap] = titleSources.find(([param]) => !!param) || [
    null,
    null,
  ]

  // if the title maps to productClusterIds, gets the cluster name
  // from the product list (which hopefully at least one
  // product will have listed on its productClusters key)
  if (titleMap === 'productClusterIds') {
    const productCluster = getClusterById(products, titleValue)

    if (productCluster) {
      title = productCluster
    }
  } else {
    title = titleValue
  }

  if (!title) {
    return null
  }

  return (
    <h1 className={classNames(styles.galleryTitle, 't-heading-1')}>
      {decodeURI(title)}
    </h1>
  )
}

export default SearchTitle
