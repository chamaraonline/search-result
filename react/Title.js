import classNames from 'classnames'
import React from 'react'

import styles from './searchResult.css'

const Title = ({ params, showTitle }) => {
  if (!showTitle) {
    return null
  }

  const title = params.term ||
    params.subcategory ||
    params.category ||
    params.department

  return (
    <h1 className={classNames(styles.galleryTitle, 't-heading-1')}>
      {title}
    </h1>
  )
}

export default Title
