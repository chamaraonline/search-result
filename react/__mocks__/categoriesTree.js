export default [
  {
    Id: 1,
    Name: 'Eletrônicos',
    Link: '/Eletronicos',
    Quantity: 0,
    Children: [
      {
        Id: 2,
        Name: 'Smartphones',
        Link: '/Eletronicos/Smartphones',
        Quantity: 0,
      },
      {
        Id: 3,
        Name: 'Videogames',
        Link: '/Eletronicos/Videogames',
        Quantity: 0,
      },
      {
        Id: 4,
        Name: 'TVs',
        Link: '/Eletronicos/TVs',
        Quantity: 0,
      },
    ],
  },
  {
    Id: 5,
    Name: 'Livros',
    Link: '/Livros',
    Quantity: 0,
    Children: [
      {
        Id: 6,
        Name: 'HQs e Mangas',
        Link: '/Livros/HQs e Mangas',
        Quantity: 0,
      },
    ],
  },
  {
    Id: 7,
    Name: 'Roupas',
    Link: '/Roupas',
    Quantity: 0,
    Children: [
      {
        Id: 8,
        Name: 'Blusas',
        Link: '/Roupas/Blusas',
        Quantity: 0,
      },
      {
        Id: 9,
        Name: 'Calças',
        Link: '/Roupas/Calcas',
        Quantity: 0,
      },
      {
        Id: 10,
        Name: 'Moletom',
        Link: '/Roupas/Moletom',
        Quantity: 0,
      },
    ],
  },
]

export const numberOfFilters = 10
