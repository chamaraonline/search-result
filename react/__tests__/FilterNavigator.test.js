/* eslint-env jest */
import React from 'react'
import { render } from '@vtex/test-tools/react'

import categoriesTree from 'categoriesTree'

import FilterNavigator from '../FilterNavigator'
import QueryContext from '../components/QueryContext'

describe('<FilterNavigator />', () => {
  const renderComponent = (customProps = { query: 'clothing' }) => {
    const props = {
      map: 'c',
      tree: categoriesTree,
      ...customProps,
    }

    return render(
      <QueryContext.Provider
        value={{ query: customProps.query, map: props.map }}
      >
        <FilterNavigator {...props} />
      </QueryContext.Provider>
    )
  }

  it('should match snapshot with all', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })

  it('should display related categories', () => {
    const { asFragment } = renderComponent({ query: 'Livros' })

    expect(asFragment()).toMatchSnapshot()
  })
})
