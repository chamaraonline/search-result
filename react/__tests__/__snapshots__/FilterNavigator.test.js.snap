// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterNavigator /> should display related categories 1`] = `
<DocumentFragment>
  <div
    class="filters"
  >
    <div
      class=""
    >
      <div
        class="filter pv3 bb b--muted-4 filterAvailable"
      >
        <div
          class="pointer"
        >
          <div
            class="filterTitle t-heading-7 flex items-center justify-between"
          >
            Categories
            <span
              class="filterIcon"
              style="height: 10px;"
            >
              <div>
                IconCaret
              </div>
            </span>
          </div>
        </div>
        <div
          class="pt2 overflow-y-auto"
          style="max-height: 200px;"
        >
          <div
            style="overflow: hidden; height: 0px;"
          >
            <div
              class="filterContent"
            >
              <a
                class="mockedLink"
              >
                <span
                  class="t-small c-muted-1 bb db"
                  style="border-color: transparent; border-width: 1px;"
                >
                  Livros
                </span>
              </a>
              <a
                class="mockedLink"
              >
                <span
                  class="t-small c-muted-1 bb db"
                  style="border-color: transparent; border-width: 1px;"
                >
                  HQs e Mangas
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<FilterNavigator /> should match snapshot with all 1`] = `
<DocumentFragment>
  <div
    class="filters"
  >
    <div
      class=""
    >
      <div
        class="filter pv3 bb b--muted-4 filterAvailable"
      >
        <div
          class="pointer"
        >
          <div
            class="filterTitle t-heading-7 flex items-center justify-between"
          >
            Categories
            <span
              class="filterIcon"
              style="height: 10px;"
            >
              <div>
                IconCaret
              </div>
            </span>
          </div>
        </div>
        <div
          class="pt2 overflow-y-auto"
          style="max-height: 200px;"
        >
          <div
            style="overflow: hidden; height: 0px;"
          >
            <div
              class="filterContent"
            >
              <a
                class="mockedLink"
              >
                <span
                  class="t-small c-muted-1 bb db"
                  style="border-color: transparent; border-width: 1px;"
                >
                  Eletrônicos
                </span>
              </a>
              <a
                class="mockedLink"
              >
                <span
                  class="t-small c-muted-1 bb db"
                  style="border-color: transparent; border-width: 1px;"
                >
                  Smartphones
                </span>
              </a>
              <a
                class="mockedLink"
              >
                <span
                  class="t-small c-muted-1 bb db"
                  style="border-color: transparent; border-width: 1px;"
                >
                  Videogames
                </span>
              </a>
              <a
                class="mockedLink"
              >
                <span
                  class="t-small c-muted-1 bb db"
                  style="border-color: transparent; border-width: 1px;"
                >
                  TVs
                </span>
              </a>
              <a
                class="mockedLink"
              >
                <span
                  class="t-small c-muted-1 bb db"
                  style="border-color: transparent; border-width: 1px;"
                >
                  Livros
                </span>
              </a>
              <a
                class="mockedLink"
              >
                <span
                  class="t-small c-muted-1 bb db"
                  style="border-color: transparent; border-width: 1px;"
                >
                  HQs e Mangas
                </span>
              </a>
              <a
                class="mockedLink"
              >
                <span
                  class="t-small c-muted-1 bb db"
                  style="border-color: transparent; border-width: 1px;"
                >
                  Roupas
                </span>
              </a>
              <a
                class="mockedLink"
              >
                <span
                  class="t-small c-muted-1 bb db"
                  style="border-color: transparent; border-width: 1px;"
                >
                  Blusas
                </span>
              </a>
              <a
                class="mockedLink"
              >
                <span
                  class="t-small c-muted-1 bb db"
                  style="border-color: transparent; border-width: 1px;"
                >
                  Calças
                </span>
              </a>
              <a
                class="mockedLink"
              >
                <span
                  class="t-small c-muted-1 bb db"
                  style="border-color: transparent; border-width: 1px;"
                >
                  Moletom
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
